src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -608,7 +609,31 @@ case class IndexLogEntryTag[T](name: String)
 object IndexLogEntry {
   val VERSION: String = "0.1"
 
+  val HYPERSPACE_PROJECT_VERSION: String = "hyperspaceVersion"
+
   def schemaString(schema: StructType): String = schema.json
+
+  /*
+   * Use this method to create a new IndexLogEntry. Which automatically includes
+   * all common default hyperspace project properties.
+   * TODO: force dev to use this method as this takes into account all
+   *  system properties that needed to be added by default. Currently, dev can also
+   *  create IndexLogEntry from case class.
+   *  https://github.com/microsoft/hyperspace/issues/370
```

Comment:
Can you add a similar comment to the `IndexLogEntry` constructor and we need to add `require` when there is a breaking change?